cli/generator/generate.js

```diff
@@ -0,0 +1,54 @@
+const chalk = require('chalk')
+const cliInput = require('./cliInput')
+const RootDirController = require('./RootDirController')
+const packageMerge = require('./packageMerge')
+const exec = require('child-process-promise').exec
+const depcheck = require('depcheck')
+const path = require("path")
+
+module.exports = {
+  run: async () => {
+    const projectDir = cliInput.get('projectDirectory')
+    const rootDir = new RootDirController(projectDir)
+    let peerDepInstall
+    // Create directory for new implementing application
+    if (!rootDir.has()) rootDir.make()
+
+    return new Promise( async (resolve, reject) => {
+      // Clone ol-kit directory into new implementing application
+      console.log(chalk.cyan(`    Cloning ol-kit files into /${projectDir}`))
+      return exec(`git clone https://github.com/MonsantoCo/ol-kit.git ./${projectDir}`) 
+      .then(() => {
+        // If our peer dependencies ship a breaking change this will need to be addressed.
+        const peerDependencies = require(`${rootDir.projectPath}/package.json`).peerDependencies
+        peerDepInstall = Object.keys(peerDependencies).join(' ')
+        return peerDepInstall
+      })
+      .then((peerDepInstall) => {
+        console.log(chalk.cyan(`    Installing dependencies...(This may take a while)`))
+        // remove everything except app/demos/world/*
+        return exec(`cd ./${projectDir} && 
+        find . -depth -mindepth 1 \! -path "./app/demos/world/*" \! -path "./app/demos/world" -delete && 
+        mv app/demos/world/App.js . && 
+        rm -rf app && npm init -y &&
+        npm install ${peerDepInstall} --save`)
```

Comment:
Does this assume that all dependencies are runtime (and thus not do saveDev where appropriate)? Seems like it might be some unintentional package bloat.

Reply:
in this case, all the `peerDependencies` will be required by `ol-kit` at runtime and need to be installed with `--save`